tests/Makefile

```diff
@@ -11,14 +11,16 @@ SUBDIRS := $(wildcard $(addprefix $(tests.path), /testdata/template_and_smoke/*)
 $(foreach dir,$(SUBDIRS),$(eval $(call INCLUDE_FILE, $(dir))))
 
 smoke-tests.run: smoke-tests.images
-	cd $(tests.path) && go test -v -tags=smoke
+	cd $(tests.path) && go test -v -tags=smoke -count=1
 
 tests smoke-tests .PHONY: smoke-tests.run
 
 template-tests.run: template-tests.images
-	cd $(tests.path) && go test -v -tags=template
+	cd $(tests.path) && go test -v -tags=template -count=1
 
 tests template-tests .PHONY: template-tests.run
 
-integration: $(builder.image.out) $(PACK)
-	cd $(tests.path) && go test ./integration/... -v -args -pack=$(PACK) -builder=$(shell cat $(builder.image.out))
+integration-tests.run: buildpacks.images.local builder.image.local $(builder.image.out) $(PACK)
```

Comment:
This will clean out any local builder/buildpack artifacts, then build local copies (using `pull-policy if-not-present` when building builder) before running tests

Reply:
If the local builder depended on local buildpack, then we wouldn't need to specify the local buildpacks here

Reply:
That's true. I wasn't sure what our interpretation of local builder was here. It could either:

- Build a local copy and use any existing buildpacks on the daemon (i.e. "shallow" local)
- Build a local copy, ensuring the buildpacks are rebuilt from scratch beforehand (i.e. "deeply" local)

I think that the second option makes the most sense, so agree it would be a good idea to bake that idea into the local builder target itself 👍

Reply:
I think it's worth reusing any existing buildpacks, let's say I am only working on the Python buildpack. Rebuilding the Java buildpack every time will make my iterations longer! This gets longer and longer when more buildpacks are supported.

If I'm testing Python, I could do a `make buildpacks.python.clean integration-tests.run`, or something along those lines. If I truly want to rebuild all dependencies, you can run `make` with the `-B` flag to always rebuild everything.

Reply:
I'm a little concerned that the default behavior for `make integration-tests` will become unintuitive if we don't clean anything, as a dev could make breaking buildpack changes, run `make integration-tests` and falsely think that their code is good. You have a great point about the efficiency tradeoff though.

Reply:
That should also be the case if the buildpack depends on the invoker!

Reply:
It nearly worked for the invoker, but I had to make some adjustments. More in this thread: https://github.com/vmware-tanzu/function-buildpacks-for-knative/pull/140/files#r990534272